tpm.c

```diff
@@ -108,6 +108,45 @@ static EFI_STATUS tpm_locate_protocol(efi_tpm_protocol_t **tpm,
 	return EFI_NOT_FOUND;
 }
 
+static EFI_STATUS cc_log_event_raw(EFI_PHYSICAL_ADDRESS buf, UINTN size,
+				   UINT8 pcr, const CHAR8 *log, UINTN logsize,
+				   UINT32 type, CHAR8 *hash)
+{
+	EFI_STATUS efi_status;
+	EFI_CC_EVENT *event;
+	efi_cc_protocol_t *cc;
+	EFI_CC_MR_INDEX mr;
+
+	efi_status = LibLocateProtocol(&EFI_CC_MEASUREMENT_PROTOCOL_GUID,
+				       (VOID **)&cc);
+	if (EFI_ERROR(efi_status) || cc == NULL)
+		return EFI_SUCCESS;
+
+	efi_status = cc->map_pcr_to_mr_index(cc, pcr, &mr);
+	if (EFI_ERROR(efi_status))
+		return EFI_NOT_FOUND;
+
+	UINTN event_size = sizeof(*event) - sizeof(event->Event) + logsize;
+
```

Comment:
... the math here becomes simpler here.

(didn't we have the same discussion on the grub patch already?)

Reply:
I am dilemma since TPM structure has similiar issue at https://github.com/rhboot/shim/blob/75449bcab1aad6f32046531a5b47ed3dc14cea6b/include/tpm.h#L39 and https://github.com/rhboot/shim/blob/75449bcab1aad6f32046531a5b47ed3dc14cea6b/include/tpm.h#L138 Do you want I fixed TPM and CC protocol at same time? At least, they should be consistent I think. What about your thoughts?